### PR TITLE
Extract scheduling and NWD toggle into components

### PR DESCRIPTION
**Pull request by oliverguenther**
This was planned for the initial breakup of single and multi modal, but was forgotten.

Fixes https://community.openproject.org/wp/44119

### CODE REVIEW

File: frontend/src/app/spot/components/toggle/toggle.component.ts [hunk 1]
```diff
@@ -36,8 +37,13 @@ export class SpotToggleComponent<T> implements ControlValueAccessor {
 
   @Input() public value:T;
 
+  constructor(
+    private cdRef:ChangeDetectorRef,
+  ) {}
+
   writeValue(value:T):void {
     this.value = value;
+    this.cdRef.markForCheck();
```
Comment (oliverguenther) [inline frontend/src/app/spot/components/toggle/toggle.component.ts:46]: This fixes the CVA failing to be used in OnPush situations. It requires to be marked for checked unless an input binding changes in the template, which is not the case here.

File: frontend/src/app/shared/components/datepicker/single-date-modal/single-date.modal.html [hunk 2]
```diff
@@ -10,37 +10,14 @@
...
-            [name]="'include-non-working-days'"
-            [(ngModel)]="ignoreNonWorkingDays"
-            (ngModelChange)="changeNonWorkingDays()"
-            data-qa-selector="op-datepicker-modal--include-non-working-days"
-          ></spot-toggle>
-        </div>
-      </div>
+      <op-datepicker-scheduling-toggle
+        [scheduleManually]="scheduleManually"
+        (schedulingChanged)="scheduleManually = $event"
```
Comment (b12f) [inline frontend/src/app/shared/components/datepicker/single-date-modal/single-date.modal.html]: It's a rather big change but it'd be nice if this worked with ngModel as a ControlValueAccessor

**Review by b12f — APPROVED** [review]
All right for me like this.

File: frontend/src/app/shared/components/datepicker/single-date-modal/single-date.modal.html (hunk 2 above)
Comment (oliverguenther) [reply frontend/src/app/shared/components/datepicker/single-date-modal/single-date.modal.html]: It's a bit weird to have to repeat the CVA (as it's simply forwarding to spot-toggle anyway), but I agree that it makes using them a bit nicer. Done

**Pull request merged**